### PR TITLE
Increase test coverage by including more modern HHVMs

### PR DESCRIPTION
**Pull request by lexidor**
Testing on hhvm 4.0.4 is not sufficient for people who are closer to master. Increasing coverage by running tests on more environments.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ language: generic
 services: docker
 env:
   - HHVM_VERSION=4.0.4
```
Comment (fredemmott) [inline .travis.yml]: This one should probably be 4.0-latest too

**Comment by fredemmott** [comment]
I think you need to delete the composer.lock and add it to .gitignore for this to work; this is a tradeoff though :(

Another approach is to create hhvm-4.0.Composer.lock files etc and make travis switch between them; FB's tried that in the past for our open source projects and generally found it more painful than no lock file

**Comment by lexidor** [comment]
All the errors are because of broken dependencies. That stinks.

**Comment by fredemmott** [comment]
- Upgrade seems worse than removing the lockfile
- The lockfile was already removed for -dev builds, which means failures show on the nightly ; the hacktest dependency in composer.json needs to be updated to allow either 1.0 or 2.0

**Comment by lexidor** [comment]
## This branch will conflict *heavily* with #25 and I would prefer if that one could be merged first.

**Comment by lexidor** [comment]
I would love to show you what the newly generated source looks like, but the codegen in hhvm 4.32-dev is *really* different. `git diff | wc 1319    5763   57893` It also removed some HH_FIXME's that might still be needed by some HHVM version that we still support, so I can't generate the code for us all.

**Comment by lexidor** [comment]
You can see the diff here: https://github.com/lexidor/hack-json-schema/pull/1 (DO NOT MERGE)

**Comment by lexidor** [comment]
I am aware of the new lint errors. I'll address them sometime during this work week.

**Pull request merged**